src/arrayUtils.js

```diff
@@ -14,9 +18,7 @@ function intArrayToString(array) {
   for (var i = 0; i < array.length; i++) {
     var chr = array[i];
     if (chr > 0xFF) {
-#if ASSERTIONS
-      assert(false, 'Character code ' + chr + ' (' + String.fromCharCode(chr) + ')  at offset ' + i + ' not in 0x00-0xFF.');
-#endif
+      maybeAssert(false, 'Character code ' + chr + ' (' + String.fromCharCode(chr) + ')  at offset ' + i + ' not in 0x00-0xFF.');
```

Comment:
The real problem is here though. This adds a function call at runtime for every iteration in the loop, which if it isn't optimized by the JIT, will make `intArrayToString` much slower, and we do care about speed there.

The only way around that is preprocessing, which we now want to avoid - so we need something else creative to get around this. Maybe just two copies of the code wouldn't be too bad.